Fix: Preserve canvas position when calling noSmooth() in WEBGL

ImRAJAS-SAMSE opened the pull request:
<!--
  Thank you for contributing! Please use this pull request (PR) template.

  In the description field of this PR, include "resolves #XXXX" tagging the issue you are fixing. If this PR addresses the issue but doesn't completely resolve it (i.e., the issue should remain open after merging), write "addresses #XXXX".
-->

### **🔧 Resolves #7548**  

## **Changes:**
- **Prevents unnecessary canvas recreation** in `p5.RendererGL.js` by ensuring `noSmooth()` does not trigger a full reset in WebGL mode.  
- **Preserves canvas positioning** when calling `noSmooth()`, avoiding unintended resets when `position(x, y)` is used.  

## **Screenshots of the Change:**  
<!-- If applicable, add before/after screenshots showcasing the fix. -->  
✅ *The canvas now maintains its position correctly after calling `noSmooth()`.*  

---

### **✅ PR Checklist**
- [x] `npm run lint` passes  
- [x] [Inline reference] is included / updated  
- [x] [Unit tests] are included / updated  

[Inline reference]: https://p5js.org/contribute/contributing_to_the_p5js_reference/  
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

ImRAJAS-SAMSE commented:
Hi @davepagurek,  

I've rebased my changes onto `dev-2.0` and ran `npm test`. Most of the tests have passed, but I encountered **7 WebGL test failures**, including a **visual test mismatch**.  

Since my fix involves preventing unnecessary canvas recreation in WebGL mode, some tests might need updating. Should I update the reference screenshots or investigate further?  

The **other tests and linting checks are successful**. Let me know how you'd like to proceed!  

Thanks!

ImRAJAS-SAMSE commented on src/webgl/p5.RendererGL.js:
```diff
@@ -930,15 +930,22 @@ class RendererGL extends Renderer {
     const h = this.height;
     const defaultId = this.canvas.id;
-    const isPGraphics = this._pInst instanceof Graphics;
+    const isPGraphics = this._pInst instanceof p5.Graphics;
```
Got it! I'll turned off my auto-formatter and will remove the `p5.` prefix, including the one in `Element.call(...).` Let me know if there's anything else I should adjust. Thanks for the review.

ImRAJAS-SAMSE commented:
Hi @davepagurek,

All tests have now passed, and I’ve turned off my auto-formatter to prevent unnecessary diffs in future too. Let me know if anything else needs adjustment. Thanks!

davepagurek commented on src/webgl/p5.RendererGL.js:
```diff
@@ -932,13 +932,20 @@ class RendererGL extends Renderer {
       const node = pg._pInst._userNode || document.body;
       node.appendChild(pg.canvas);
-      Element.call(pg, pg.canvas, pg._pInst);
+      p5.Element.call(pg, pg.canvas, pg._pInst);
```
Does this still work if we take off the p5 prefix here too as it was before?

```suggestion
      Element.call(pg, pg.canvas, pg._pInst);
```

ImRAJAS-SAMSE replied:
Yes, I tested it after removing the p5. prefix, and everything works correctly. The canvas still preserves its position after calling noSmooth().

davepagurek reviewed (commented):
Thanks, I'm glad to see tests are working now! The last thing I think we'll want to do is to add a unit test to make sure that future changes to p5 won't accidentally break this change without us knowing. We could add a test that changes the position of a webgl canvas, then calls noSmooth, and then checks to see that the new element has the same position style attributes as before?

Let me know if I can help explain anything about unit testing in p5!

ImRAJAS-SAMSE commented:
Added a unit test to verify that noSmooth() preserves canvas position in WebGL. Also added comments in p5.RendererGL.js for clarity. All tests passed locally, but one CI test is still failing.

davepagurek commented on test/unit/webgl/test_noSmoothCanvasPosition.js:
```diff
@@ -0,0 +1,26 @@
+import { describe, it, expect } from 'vitest';
+
+// Importing p5 
+import '../../../lib/p5.js';
```
I think this import is giving it trouble because `lib` is only built on release or when manually built.

Other test files tend to import p5 differently (see for example `test/unit/webgl/RendererGL.js`.) It might be easier (and maybe more organized too) to actually just put this test case into the existing `RendererGL` test file, as then you can just use the `myp5` variable it has already imported and created.

ImRAJAS-SAMSE replied:
Got it! I'll move the test into p5.RendererGL.js and reuse the existing myp5 instance. Thanks for the suggestion!

ImRAJAS-SAMSE commented:
Moved the noSmooth() canvas position test to` p5.RendererGL.js` as suggested and removed the old test file. All tests and lint checks have passed. Are there any more adjustments needed. Thanks

davepagurek reviewed (approved):
Thanks @ImRAJAS-SAMSE, everything looks good to go!

davepagurek commented:
@all-contributors please add @ImRAJAS-SAMSE for code
